### PR TITLE
[ios][sqlite] Migrate to Expo Modules API

### PR DESCRIPTION
**Pull request by alanjhughes**
# Why
Continue migrating modules to Expo Modules API

# How
Followed the usual migration steps

# Test Plan
All tests passing in `bare-expo`

### CODE REVIEW

File: packages/expo-sqlite/ios/SQLiteModule.swift [hunk 1]
```diff
@@ -0,0 +1,209 @@
...
+      var stringArg: NSString
+
+      if arg is NSString {
+        stringArg = NSString(format: "%@", arg)
+      } else {
+        stringArg = arg.description as NSString
+      }
+
+      let data = stringArg.data(using: NSUTF8StringEncoding)
+      sqlite3_bind_text(statement, index, stringArg.utf8String, -1, nil)
```
Comment (Kudo) [inline packages/expo-sqlite/ios/SQLiteModule.swift:187]: is it intentional to pass `-1, nil`?

reference for original code:
https://github.com/expo/expo/blob/6afa4eb4d32197682433fab743949aa872d431d2/packages/expo-sqlite/ios/EXSQLite/EXSQLite.m#L252

**Review by Kudo — COMMENTED** [review]
i'm leaving some comments and questions. thanks @alanjhughes for helping the migration as always.

File: packages/expo-sqlite/ios/SQLiteModule.swift (hunk 1 above)
Comment (alanjhughes) [reply packages/expo-sqlite/ios/SQLiteModule.swift:187]: Yes, this was intentional. A negative value means bytes up to the first terminating 0. This seems to be the convention when using this API in Swift. I could do this if you prefer.
```swift
sqlite3_bind_text(statement, index, stringArg.utf8String, Int32(data?.count ?? 0), nil)
```

SQLITE_TRANSIENT is not imported in swift so it is passed as nil. I could also define it myself and use it that way
```swift
let SQLITE_TRANSIENT = unsafeBitCast(OpaquePointer(bitPattern: -1), to: sqlite3_destructor_type.self)
sqlite3_bind_text(statement, index, stringArg.utf8String, Int32(data?.count ?? 0), SQLITE_TRANSIENT)
```

File: packages/expo-sqlite/ios/SQLiteModule.swift [hunk 2]
```diff
@@ -0,0 +1,211 @@
...
+      return [error]
+    }
+
+    return [nil, insertId, rowsAffected, columnNames, resultRows]
+  }
+
+  private func bindStatement(statement: OpaquePointer?, with arg: NSObject, at index: Int32) {
+    if arg == NSNull() {
+      sqlite3_bind_null(statement, index)
+    } else if arg is NSInteger {
```
[the remark this answers is not in the document]
Comment (alanjhughes) [reply packages/expo-sqlite/ios/SQLiteModule.swift:175]: No problem, I'll change it back 👍

[the remark this answers is not in the document]
Comment (Kudo) [reply packages/expo-sqlite/ios/SQLiteModule.swift]: thanks @alanjhughes. we are good to go. let's ship it. appreciated for your help again ❤️

**Review by Kudo — APPROVED** [review]

File: packages/expo-sqlite/ios/SQLiteModule.swift (hunk 2 above)
[the remark this answers is not in the document]
Comment (alanjhughes) [reply packages/expo-sqlite/ios/SQLiteModule.swift]: Thanks @Kudo happy to help 👍

**Pull request merged**